Sources/App/Scenes/CarPlaySceneDelegate.swift

```diff
@@ -13,114 +13,55 @@ public protocol EntitiesStateSubscription {
 class CarPlaySceneDelegate: UIResponder {
     private var interfaceController: CPInterfaceController?
     private var entities: HACache<Set<HAEntity>>?
-    private var domainsListTemplate: DomainsListTemplate?
-    private var serverId: Identifier<Server>?
 
-    private let carPlayPreferredServerKey = "carPlay-server"
-
-    private func setServer(server: Server) {
-        serverId = server.identifier
-        prefs.set(server.identifier.rawValue, forKey: carPlayPreferredServerKey)
-        setDomainListTemplate(for: server)
-        updateServerListButton()
-    }
-
-    private func updateServerListButton() {
-        domainsListTemplate?.setServerListButton(show: Current.servers.all.count > 1)
+    private var domainsListTemplate: CarPlayTemplateProvider?
+    private var serversListTemplate: CarPlayTemplateProvider?
+    private var actionsListTemplate: CarPlayTemplateProvider?
+
+    @objc private func updateTemplates() {
```

Comment:
feels like a lot of this doesn't necessarily need to reset each time. we're not really treating them as immutable, so there's a quasi-immutable thing going on here. maybe it would be better to strongly declare the template providers on init -- when changing server, tell them about which one is selected (if they care); but otherwise let them update responsively.

Reply:
Agree, I've done some clean up